backend/tests/api/user_test.go

```diff
@@ -38,3 +41,93 @@ func TestGetAllUsersWorks(t *testing.T) {
 	// assert that the user returned from the database is the same as the user returned from the API
 	assert.Equal(dbUser, respUser)
 }
+
+func TestGetUserHappyPath(t *testing.T) {
+	// initialize the test
+	app, assert := InitTest(t)
+
+	// create a GET request to the APP/api/v1/users/:id endpoint
+	req := httptest.NewRequest("GET", fmt.Sprintf("%s/api/v1/users/1", app.Address), nil)
+
+	// send the request to the app
+	resp, err := app.App.Test(req)
+
+	assert.NilError(err)
+
+	assert.Equal(200, resp.StatusCode)
+
+	// decode the response body into a slice of users
+	var user *models.User
+
+	err = json.NewDecoder(resp.Body).Decode(&user)
+
+	assert.NilError(err)
+
+	dbUser, err := transactions.GetUser(app.Conn, "1")
+
+	assert.NilError(err)
+
+	assert.Equal(dbUser.Email, user.Email)
+	assert.Equal(user.Email, "generatesac@gmail.com")
+}
+
+func TestGetUserBadRequest(t *testing.T) {
```

Comment:
can we split this into two tests because right now this has two purpose statements.

1. test that get user returns a 400 for random string input
2. test that get user returns a 400 for null input